controllers/machinehealthcheck_controller.go

```diff
@@ -187,36 +188,33 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, cluster *c
 	// Create client for target cluster
 	clusterClient, err := remote.NewClusterClient(ctx, r.Client, util.ObjectKey(cluster), r.scheme)
 	if err != nil {
-		logger.Error(err, "Error building target cluster client")
-		return ctrl.Result{}, err
+		return ctrl.Result{}, errors.Wrapf(err, "Error building target cluster client")
```

Comment:
One of the reasons that we've used the logger here in the past is to have the output printed using the structured logging information, if we just wrap and return then we are losing some of that structured logging information. It looks like in this case, we would lose the `WithValues("cluster", cluster.Name)` compared to the logger being used in `Reconcile()`.

Similar for other logging changes below.

Reply:
Yeah this was from https://github.com/kubernetes-sigs/cluster-api/pull/3108#discussion_r434094334 @vincepri

Reply:
I don't actually know that logging the cluster name is that important though. You could get it as long as you know what MHC it was.

Reply:
We could also add it to the context in `Reconcile` since we actually fetch it there too

Reply:
I think it's good to have, since it avoids needing to do a separate lookup during troubleshooting. +1 to adding it to the logger in `Reconcile`